lib/bowling.rb

```diff
@@ -0,0 +1,39 @@
+class Bowling
+  attr_reader :total_score, :frame
+
+  def initialize
+    @total_score = 0
+  end
+
+  def total_frames(frame_scores)
+    raise 'Minimum of 10 frames must be entered' if frame_scores.count < 10
+    # p frame_scores
+    frame_scores.map.with_index do |frame, index|
```

Comment:
Nice use of map_with_index. This is the beginnings of functional programming. Well worth reading up on it. The better you get with this style of programming the cleaner and more readable your code will become